addons/microsoft_calendar/models/calendar.py

```diff
@@ -171,12 +171,15 @@ def write(self, values):
                 self._forbid_recurrence_update()
 
         # When changing the organizer, check its sync status and verify if the user is listed as attendee.
-        if values.get('user_id') and self.user_id.id != values['user_id']:
-            sender_user, partner_ids = self._get_organizer_user_change_info(values)
-            partner_included = sender_user.partner_id in self.attendee_ids.partner_id or sender_user.partner_id.id in partner_ids
-            self._check_organizer_validation(sender_user, partner_included)
-            self._recreate_event_different_organizer(values, sender_user)
-            values['active'] = False
+        # The organizer change is only valid for regular events as recurrences are not editable from Odoo.
+        for event in self:
+            if values.get('user_id') and event.user_id.id != values['user_id']:
+                sender_user, partner_ids = event._get_organizer_user_change_info(values)
+                partner_included = sender_user.partner_id in event.attendee_ids.partner_id or sender_user.partner_id.id in partner_ids
+                if not event.recurrency and not event.recurrence_id:
+                    event._check_organizer_validation(sender_user, partner_included)
+                    event._recreate_event_different_organizer(values, sender_user)
+                    values['active'] = False
```

Comment:
you change the active value for all events even if only one is impacted. It seems weird.